@x10an14 That type-hinting commit didn't really work. From a virtualenv after pip -r requirements.txt:

```
(code) [root@3c5073631c90 source]# python3.4 ./app.py
Traceback (most recent call last):
  File "./app.py", line 34, in <module>
    def ping(host: string=None):
NameError: name 'string' is not defined
```

Now, I realize that I need to sit down and get more into python3.4+, as I wasn't sure how to fix it, so I just reverted it for now. Runs fine without it. Will test on Openshift as well before merging.

--- a/app.py
+++ b/app.py
@@ -44,7 +44,7 @@ def ping(host=None):
     #   (It's a 4-packet ping for crying out loud...)
     cmd, time_limit = "ping -c 4".split(" "), 16
     proc = subprocess.Popen(
-        args=cmd.append(host),
+        args=cmd.append(host) if host else cmd,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
         universal_newlines=True)

--- a/app.py
+++ b/app.py
@@ -29,8 +29,8 @@ def index():
     return redirect("/ping", code=302)
 
 
-@app.route('/ping')
-@app.route('/ping/<host>')
+@app.route('/ping/')
+@app.route('/ping/<host>/')
 def ping(host=None):
     """Main function where the 'magic' happens!..."""
     if not host:

--- a/app.py
+++ b/app.py
@@ -42,9 +42,9 @@ def ping(host: string=None):
 
     # Default time_limit of hardcoded 16 seconds.
     #   (It's a 4-packet ping for crying out loud...)
-    cmd, time_limit = "ping -c 4".split(" "), 16
+    time_limit, cmd = 16, ["ping", "-c", "4", str(host)]
     proc = subprocess.Popen(
-        args=cmd.append(host),
+        args=cmd,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
         universal_newlines=True)

